Tests/Functional/DependencyResolutionTests.swift

```diff
@@ -22,6 +22,12 @@ class DependencyResolutionTestCase: XCTestCase {
         }
     }
 
+    func testInternalExecAsDep() {
+        fixture(name: "DependencyResolution/Internal/InternalExecutableAsDependency") { prefix in
```

Comment:
Do we need to invoke `swift build` to test it? Wouldn't it be better to instantiate a package `let package = Package(...)` with executable dependency, call `package.modules()` method and check return type or catch exception. 

That would be a more isolated unit test and it would be way faster.